### PR TITLE
Added fields to system-user assertion UI to limit it to a set of serial-numbers

### PR DESCRIPTION
**Pull request by glower**

### CODE REVIEW

**Comment by glower** [comment]
[https://drive.google.com/file/d/18NvXbt_NxnxClwbfUh28e6SfJtnRNH3A/view?usp=sharing](UI screenshot 1)
[https://drive.google.com/file/d/1Sn7d1CzrLyLc_DypP6TIRm2qflNGeVAF/view?usp=sharing](UI screenshot 2)

File: webapp-admin/src/components/SystemUserForm.js [hunk 1]
```diff
@@ -188,6 +202,20 @@ class SystemUserForm extends Component {
                             })}
                             </select>
                         </label>
+                        <label>Limit this system-user to a set of serial numbers:&nbsp;
+                            <button className="p-button--brand" title="Add serial number" onClick={this.handleAddEmptySerialField}>
+                                <i className="fa fa-plus"></i>
```
Comment (anthonydillon) [inline webapp-admin/src/components/SystemUserForm.js:207]: If you are using Vanilla here we have a plus icon: `p-icon--plus`. https://vanillaframework.io/docs/patterns/icons

File: webapp-admin/src/components/SystemUserForm.js [hunk 2]
```diff
@@ -100,17 +101,30 @@ class SystemUserForm extends Component {
         this.setState({until: date});
     }
 
+    handleAddEmptySerialField = (e) => {
+        e.preventDefault()
+        this.setState({serials: this.state.serials.concat([''])})
+    }
+
+    handleAddSerialNumber = (e) => {
+        const i = e.target.dataset["index"];
```
Comment (anthonydillon) [inline webapp-admin/src/components/SystemUserForm.js:110]: You can just do `e.target.dataset.index;`

File: webapp-admin/src/components/SystemUserForm.js [hunk 3]
```diff
@@ -100,17 +101,30 @@ class SystemUserForm extends Component {
         this.setState({until: date});
     }
 
+    handleAddEmptySerialField = (e) => {
+        e.preventDefault()
+        this.setState({serials: this.state.serials.concat([''])})
```
Comment (anthonydillon) [inline webapp-admin/src/components/SystemUserForm.js:106]: I would suggest creating a `tmpSerials` simular to `handleAddSerialNumber` and just push an empty string to it:
```js
let tmpSerials = this.state.serials;
tmpSerials.push('');
this.setState({serials: tmpSerials});  
```

**Review by anthonydillon — CHANGES_REQUESTED** [review]
Looks good, few suggestions in the code.

File: webapp-admin/src/components/SystemUserForm.js [hunk 4]
```diff
@@ -188,6 +202,20 @@ class SystemUserForm extends Component {
                             })}
                             </select>
                         </label>
+                        <label>Limit this system-user to a set of serial numbers:&nbsp;
+                            <button className="p-button--brand" title="Add serial number" onClick={this.handleAddEmptySerialField}>
```
Comment (anthonydillon) [inline webapp-admin/src/components/SystemUserForm.js:206]: I would suggest using [p-button--neutral](https://vanillaframework.io/docs/patterns/buttons#neutral) as the orange button here is to strong.

The button would also benifit from being more [dense](https://vanillaframework.io/docs/patterns/buttons#dense) and [inline](https://vanillaframework.io/docs/patterns/buttons#inline).

```html
<button className="p-button--neutral is-dense is-inline" title="Add serial number" onClick={this.handleAddEmptySerialField}>
```

Comment (glower) [reply webapp-admin/src/components/SystemUserForm.js]: That looks much better now!

**Comment by glower** [comment]
* Implemented feedback from PR
* Updated UI screenshots

**Review by anthonydillon — APPROVED** [review]
LGTM, thanks for the updates

**Review by suligap — APPROVED** [review]
The Go part LGTM. Thanks.

**Pull request merged**